### DISCUSSION TITLE
Exit by pressing

### DISCUSSION THREAD

**Discussion by Mikhail666999** [gilzoide/godot-lua-pluginscript#40] [Q&A] [answered]
Hello it's me again) This time I have a different problem. I'm trying to exit by clicking.
I tried to do so 
`self:get_tree().quit();`

godot throws this error 
`E 0:00:02.480   quit: attempt to index local 'o' (a nil value)
stack traceback:
	[string "res://assets/player/player.lua"]:17: in function <[string "res://assets/player/player.lua"]:11>
in call _physics_process @ res://assets/player/player.lua
<C source code++>lps_init_script:1344 @ quit()`

What am I wrong about?

**Comment by gilzoide** [comment, marked as answer]
The problem is probably the lack of using Lua's method notation:
```lua
self:get_tree():quit()
            -- ^ here!
```

**Reply by Mikhail666999** [reply]
зараза ощущаю себя идиотом...
I feel like an idiot...

**Reply by Mikhail666999** [reply]
thank you for the answer

**Reply by gilzoide** [reply]
> I feel like an idiot...

Please don't. Using a new language is always tricky, and it gets worse if the error messages are cryptic suchs as the one you got there.
